### DISCUSSION TITLE
How to do Basic Auth?

### DISCUSSION THREAD

**Discussion by jstcki** [vercel/vercel#3972] [General] [answered]
Hi!

I tried using the `headers` field in now.json but that didn't work.

```json
{
  "version": 2,
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "Authorization": "Basic ...."
        }
      ]
    }
  ]
}
```

I couldn't find anything in the docs – is this not possible?

In a next.js app, would adding the authorization header in _app.js `getInitialProps` work?

**Comment by jstcki** [comment]
Obviously adding the `Authorization` header doesn't work … that needs to be sent from the client 🤦‍♂ .

So I tried using `getInitialProps` on _app.js, but that does not work for SSGed pages. So I'm not  sure how to solve this?

**Comment by styfle** [comment]
Hi @herrstucki 

There isn't an easy solution for Basic Authentication today without implementing manually with a Serverless Function.

We are looking at ways of solving this problem, most likely with a toggle in the [Project Settings](https://zeit.co/blog/advanced-project-settings) to make the project password protected.

**Reply by jstcki** [reply]
Thanks for the reply @styfle. Something like this would really be useful! Due to the tight integration of Now and next.js I wouldn’t even know where to start to put a SSG next site in a serverless function 🤔 I really like how Netlify is enabling this without any code changes.

**Comment by wiiiimm** [comment, marked as answer]
Hi @herrstucki , we have just enabled the password protection and SSO protection features for Pro and Enterprise plans. Please take a look at https://vercel.com/blog/protecting-deployments for more information.

**Reply by coderbuzz** [reply]
Hi, I want to make sure is **req.headers.authorization** stripped from Hobby account?
